flexibleengine/data_source_flexibleengine_compute_instances.go

```diff
@@ -0,0 +1,351 @@
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"volume_attached": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"volume_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"is_sys_volume": {
+										Type:     schema.TypeBool,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"scheduler_hints": {
+							Type:     schema.TypeSet,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"group": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"metadata": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"tags": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func filterCloudServers(d *schema.ResourceData, servers []cloudservers.CloudServer) ([]cloudservers.CloudServer,
+	[]string) {
+	result := make([]cloudservers.CloudServer, 0, len(servers))
+	ids := make([]string, 0, len(servers))
+
+	for _, server := range servers {
+		if flavorName, ok := d.GetOk("flavor_name"); ok && flavorName != server.Flavor.Name {
+			continue
+		}
+		if iamgeId, ok := d.GetOk("image_id"); ok && iamgeId != server.Image.ID {
+			continue
+		}
+		if az, ok := d.GetOk("availability_zone"); ok && az != server.AvailabilityZone {
+			continue
+		}
+		if keypair, ok := d.GetOk("key_pair"); ok && keypair != server.KeyName {
+			continue
+		}
+		result = append(result, server)
+		ids = append(ids, server.ID)
+	}
+
+	return result, ids
+}
+
+func isSystemVolume(index string) bool {
+	if index == "0" {
+		return true
+	}
+	return false
+}
+
+func parseEcsInstanceVolumeAttachedInfo(attachList []cloudservers.VolumeAttached) []map[string]interface{} {
+	result := make([]map[string]interface{}, len(attachList))
+
+	for i, volume := range attachList {
+		result[i] = map[string]interface{}{
+			"volume_id":     volume.ID,
+			"is_sys_volume": isSystemVolume(volume.BootIndex),
+		}
+	}
+
+	return result
+}
+
+func setComputeInstancesParams(d *schema.ResourceData, servers []cloudservers.CloudServer, meta interface{}) diag.Diagnostics {
+	result := make([]map[string]interface{}, len(servers))
+
+	for i, val := range servers {
+		server := map[string]interface{}{
+			"id":                val.ID,
+			"user_data":         val.UserData,
+			"name":              val.Name,
+			"flavor_name":       val.Flavor.Name,
+			"status":            val.Status,
+			"flavor_id":         val.Flavor.ID,
+			"image_id":          val.Image.ID,
```

Comment:
**image_name** is missing

Reply:
done

Reply:
you can get the attribute from server.Metadata

Reply:
thanks, it´s done